### PR TITLE
Skip unnecessary operations if diff is less than 0

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In JobController#manageJob, when diff is less than or equal to zero, we still go making channel and creating WaitGroup.
This PR checks the diff and if the diff is <= 0, these operations are skipped directly.

-->
```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/ok-to-test
/priority backlog

**Comment by tedyu** [comment]
ping @janetkuo for review

**Comment by tedyu** [comment]
@erictune 
Can you review this ?

**Comment by tedyu** [comment]
@erictune
Please review

**Comment by tedyu** [comment]
@erictune
Please review

**Comment by tedyu** [comment]
@soltysh
Can you take a look ?

**Comment by tedyu** [comment]
@janetkuo
Please review

**Comment by tedyu** [comment]
ping @janetkuo for review

File: pkg/controller/job/job_controller.go [hunk 1]
```diff
@@ -747,65 +747,67 @@ func (jm *JobController) manageJob(activePods []*v1.Pod, succeeded int32, job *b
 			utilruntime.HandleError(fmt.Errorf("More active than wanted: job %q, want %d, have %d", jobKey, wantActive, active))
 			diff = 0
 		}
```
Comment (janetkuo) [inline pkg/controller/job/job_controller.go]: Is there any reason not to return here when `diff == 0`?

Comment (tedyu) [reply pkg/controller/job/job_controller.go]: Before returning, we should run the following (line 814)
```
        case err := <-errCh:
```

Comment (janetkuo) [reply pkg/controller/job/job_controller.go]: The `diff == 0` case won't touch `errCh`

**Comment by tedyu** [comment]
In favor of #80263

**Pull request closed without merging**